app/models/miq_expression/field.rb

```diff
@@ -29,11 +30,26 @@ def datetime?
     column_type == :datetime
   end
 
+  def to_s
+    assoc = associations.map { |a| ".#{a}" }.join
+    "#{model.name}#{assoc}-#{column}"
+  end
+
+  def reflections
+    klass = model
+    associations.collect do |association|
+      klass.reflect_on_association(association).tap do |reflection|
+        reflection or raise InvalidFieldError, self
```

Comment:
Nice, this gets rid of both of the try calls.

1. Do we want to change `ar_virtual#follow_associations` to have a similar interface?
2. Do we really want to introduce another class? will `IllegalArgumentException` suffice?
3. Add `ar_virtual_chain#follow_association_chain_with_virtual`?

 ```ruby
model.follow_association_with_virtual_chain(associations) || raise InvalidFieldError, self
```
1. I think I prefer `||` over `or` here.
2. Will all callers will catch `InvalidFieldError`? Could you change the callers so we have a feel for how this morphs the code? (just change one that happens to care if this is an invalid field)